### PR TITLE
Feign and Retrofit clients use OkHttp interceptors for logging at DEBUG

### PR DESCRIPTION
**Pull request by uschi2000**
In particular, Feign logging is disabled.

### CODE REVIEW

File: jaxrs-clients/src/main/java/com/palantir/remoting3/jaxrs/AbstractFeignJaxRsClientBuilder.java [hunk 1]
```diff
@@ -91,8 +92,7 @@
                 .client(target.wrapClient(createOkHttpClient(userAgent)))
                 .retryer(target)
                 .options(createRequestOptions())
-                .logger(new Slf4jLogger(JaxRsClient.class))
-                .logLevel(Logger.Level.BASIC)
+                .logLevel(Logger.Level.NONE)
```
Comment (gracew) [inline jaxrs-clients/src/main/java/com/palantir/remoting3/jaxrs/AbstractFeignJaxRsClientBuilder.java]: don't need this, this is the default

File: jaxrs-clients/src/main/java/com/palantir/remoting3/jaxrs/AbstractFeignJaxRsClientBuilder.java [hunk 2]
```diff
@@ -157,6 +157,9 @@ private static Decoder createDecoder(ObjectMapper objectMapper, ObjectMapper cbo
         // increase default connection pool from 5 @ 5 minutes to 100 @ 10 minutes
         client.connectionPool(new ConnectionPool(100, 10, TimeUnit.MINUTES));
 
+        // logging
+        client.addInterceptor(new HttpLoggingInterceptor(OkhttpSlf4jDebugLogger.INSTANCE));
```
Comment (gracew) [inline jaxrs-clients/src/main/java/com/palantir/remoting3/jaxrs/AbstractFeignJaxRsClientBuilder.java:161]: I think you can use the default okhttp logger by not passing in any arguments to `HttpLoggingInterceptor`.

File: okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/OkhttpSlf4jDebugLogger.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+public final class OkhttpSlf4jDebugLogger implements HttpLoggingInterceptor.Logger {
+
+    public static final OkhttpSlf4jDebugLogger INSTANCE = new OkhttpSlf4jDebugLogger();
+    private static final Logger log = LoggerFactory.getLogger(OkhttpSlf4jDebugLogger.class);
+
+    private OkhttpSlf4jDebugLogger() {}
+
+    @Override
+    public void log(String message) {
+        log.debug(message);
```
Comment (gracew) [inline okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/OkhttpSlf4jDebugLogger.java:34]: why debug?

File: jaxrs-clients/src/main/java/com/palantir/remoting3/jaxrs/AbstractFeignJaxRsClientBuilder.java (hunk 1 above)
Comment (uschi2000) [reply jaxrs-clients/src/main/java/com/palantir/remoting3/jaxrs/AbstractFeignJaxRsClientBuilder.java]: wanted to make explicit that we're making a choice here. will add a comment to clarify.

File: jaxrs-clients/src/main/java/com/palantir/remoting3/jaxrs/AbstractFeignJaxRsClientBuilder.java (hunk 2 above)
Comment (uschi2000) [reply jaxrs-clients/src/main/java/com/palantir/remoting3/jaxrs/AbstractFeignJaxRsClientBuilder.java:161]: does this play nice with Slf4j?

File: okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/OkhttpSlf4jDebugLogger.java (hunk 3 above)
Comment (uschi2000) [reply okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/OkhttpSlf4jDebugLogger.java:34]: I don't think I'd want to see okhttp-level information in standard/INFO level logging? what would you suggest?

File: jaxrs-clients/src/main/java/com/palantir/remoting3/jaxrs/AbstractFeignJaxRsClientBuilder.java (hunk 2 above)
Comment (gracew) [reply jaxrs-clients/src/main/java/com/palantir/remoting3/jaxrs/AbstractFeignJaxRsClientBuilder.java:161]: hm, I thought it did but taking another look it's java.util.Logger.

File: okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/OkhttpSlf4jDebugLogger.java (hunk 3 above)
Comment (gracew) [reply okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/OkhttpSlf4jDebugLogger.java:34]: ah yep, makes sense

**Review by gracew — APPROVED** [review]

**Pull request merged**